polyphemus/lib/etls/redcap/lib/model.rb

```diff
@@ -1,38 +1,60 @@
 module Redcap
   class Model
-    def self.create(model_name, scripts, magma_template, redcap_template, salt)
+    def self.create(project, model_name, model_config, magma_template, redcap_template, salt)
       class_name = model_name.to_s.split('_').map(&:capitalize).join
       model_class = Kernel.const_defined?(class_name) ?  Kernel.const_get(class_name) : nil
 
       raise "No model class for #{model_name}" unless model_class
 
-      model_class.new(scripts, magma_template, redcap_template, salt)
+      model_class.new(model_name, project, model_config, magma_template, redcap_template, salt)
     end
 
-    def initialize(scripts, magma_template, redcap_template, salt)
-      @scripts = scripts.map do |script|
+    attr_reader :scripts, :project
+
+    def initialize(model_name, project, config, magma_template, redcap_template, salt)
+      @model_name = model_name
+      @project = project
+      @config = config
+      @scripts = config[:scripts].map do |script|
         Redcap::Script.new(self, script, redcap_template)
       end
       @magma_template = magma_template
       @salt = salt
       @offset_days = {}
     end
 
-    def load(project)
-      project.logger.write("Attempting to load model #{name}.\n")
+    def events?
+      @config[:each] == "event"
+    end
+
+    def invert?
+      @config[:invert]
+    end
+
+    def magma_records
```

Comment:
minor nit, should this be called `existing_records` to differentiate more clearly from `records` ?